### PR TITLE
Feat: Convert /admin/notes to typescript 

### PR DESCRIPTION
**Pull request by pKorsholm**

### CODE REVIEW

**Review by srindom — CHANGES_REQUESTED** [review]

File: packages/medusa/src/api/routes/admin/notes/index.ts [hunk 1]
```diff
@@ -18,3 +19,13 @@ export default (app) => {
 
   return app
 }
+
+export class AdminNoteRes {
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/admin/notes/index.ts:23]:
```suggestion
export type AdminNotesRes = { 
```

File: packages/medusa/src/services/note.ts [hunk 2]
```diff
@@ -1,6 +1,9 @@
 import { MedusaError } from "medusa-core-utils"
```
Comment (olivermrbl) [inline packages/medusa/src/services/note.ts:1]: PR should not include a typed `NoteService`

File: packages/medusa/src/services/region.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import { MedusaError } from "medusa-core-utils"
```
Comment (olivermrbl) [inline packages/medusa/src/services/region.ts:1]: PR should not include a typed `RegionService`

File: packages/medusa/src/api/routes/admin/notes/index.ts [hunk 4]
```diff
@@ -18,3 +19,13 @@ export default (app) => {
...
+  note: Note
+}
+
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/admin/notes/index.ts:26]: I think we are missing a type `export type AdminNotesListRes` for listing notes.

**Review by olivermrbl — CHANGES_REQUESTED** [review]

File: packages/medusa/src/api/routes/admin/notes/index.ts [hunk 5]
```diff
@@ -18,3 +20,16 @@ export default (app) => {
 
   return app
 }
+
+export type AdminNoteRes = {
+  note: Note
+}
+
+export type AdminGetNotesRes = PaginatedResponse & {
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/admin/notes/index.ts:28]:
```suggestion
export type AdminNotesListRes = PaginatedResponse & {
```

File: packages/medusa/src/api/routes/admin/notes/index.ts [hunk 6]
```diff
@@ -18,3 +20,16 @@ export default (app) => {
 
   return app
 }
+
+export type AdminNoteRes = {
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/admin/notes/index.ts:24]:
```suggestion
export type AdminNotesRes = {
```

File: packages/medusa/src/api/routes/admin/notes/list-notes.ts [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  @IsOptional()
+  resource_id: string
+
+  @IsNumber()
+  @IsOptional()
+  limit?: number
+
+  @IsNumber()
+  @IsOptional()
+  offset?: number
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/admin/notes/list-notes.ts:62]:
```suggestion
  @IsNumber()
  @IsOptional()
  @Type(() => Number)
  limit?: number

  @IsNumber()
  @IsOptional()
  @Type(() => Number)
  offset?: number
```

**Review by olivermrbl — CHANGES_REQUESTED** [review]
Nitpicking, but to keep consistency throughout the codebase, we need it :)

File: packages/medusa/src/api/routes/admin/notes/create-note.ts [hunk 8]
```diff
@@ -33,27 +36,32 @@ import { MedusaError, Validator } from "medusa-core-utils"
...
+  @IsOptional()
+  resource_id: string
+
+  @IsString()
+  @IsOptional()
+  resource_type?: string
+
+  @IsString()
+  @IsOptional()
+  value?: string
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/admin/notes/create-note.ts:66]: These are not optional

File: packages/medusa/src/types/note.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import { Note } from "../models/note"
+import { User } from "../models/user"
+import { PartialPick } from "./common"
+
+export interface CreateNoteInput {
+  value?: string
+  resource_type?: string
+  resource_id?: string
```
Comment (olivermrbl) [inline packages/medusa/src/types/note.ts:8]: Same as above

**Review by olivermrbl — CHANGES_REQUESTED** [review]

File: packages/medusa/src/types/note.ts [hunk 10]
```diff
@@ -0,0 +1,26 @@
+import { Note } from "../models/note"
+import { User } from "../models/user"
+import { PartialPick } from "./common"
+
+export interface CreateNoteInput {
+  value: string
+  resource_type: string
+  resource_id: string
+  author_id: string
```
Comment (srindom) [inline packages/medusa/src/types/note.ts:9]: This should be optional in the case where the note is being created by a "system" e.g. a subscriber that calls the service layer directly.

File: packages/medusa/src/api/routes/admin/notes/get-note.ts [hunk 11]
```diff
@@ -1,8 +1,11 @@
+import NoteService from "../../../../services/note"
+
 /**
  * @oas [get] /notes/{id}
  * operationId: "GetNoteNote"
  * summary: "Get Note"
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/notes/get-note.ts:6]: Not your responsibility but, this should be `GetNotesNote` would you mind including this in this PR?

**Review by srindom — CHANGES_REQUESTED** [review]
Almost there

File: packages/medusa/src/types/note.ts (hunk 10 above)
Comment (olivermrbl) [reply packages/medusa/src/types/note.ts:9]: I take the blame for this one. I was too quick on the keyboard in the previous review when I saw the optional API payload properties 👍  Nice one!

**Review by olivermrbl — APPROVED** [review]

**Pull request merged**